contribs/gnofaucet/serve.go

```diff
@@ -189,12 +223,19 @@ func execServe(ctx context.Context, cfg *serveCfg, io commands.IO) error {
 
 	// Start throttled faucet.
 	st := newIPThrottler(defaultRateLimitInterval, defaultCleanTimeout)
+	// Create cooldown limiter
+	cooldownLimiter, err := NewCooldownLimiter(cfg.cooldownPeriod, cfg.dbPath)
+	if err != nil {
+		return fmt.Errorf("unable to create cooldown limiter, %w", err)
+	}
 	st.start(ctx)
 
 	// Prepare the middlewares
 	middlewares := []faucet.Middleware{
 		getIPMiddleware(cfg.isBehindProxy, st),
 		getCaptchaMiddleware(cfg.captchaSecret),
+		getAccountBalanceMiddleware(cli, cfg.maxBalance),
+		getGithubMiddleware(cfg.ghClientID, os.Getenv("GH_CLIENT_SECRET"), cooldownLimiter),
```

Comment:
The GH middleware should be optional

Reply:
done f984a7b50f01ad9950a0411b1762d211d1679e1f